### PR TITLE
Support identical feature names in different organisms 

### PR DESCRIPTION
**Pull request by njbooher**
This pull request addresses issue #344

I still need to look this over a bit more.

Also, I believe JBrowseFeatureViewSet still needs reworked to include organism somewhere. Possibly as part of the path.

I hereby agree to licence this and any previous contributions under
the terms of the GNU General Public License version 3 as published by
the Free Software Foundation

I have read the ``CONTRIBUTING.rst`` file and understand that
TravisCI will be used to confirm the tests and ``flake8`` style
checks pass with these changes.

I am happy be thanked by name in the ``CONTRIB.rst`` files,
and have added myself to the file as part of this pull request.
(*This acknowledgement is optional. Note we list the names sorted alphabetically.*)

### CODE REVIEW

File: machado/api/views.py [hunk 1]
```diff
@@ -369,6 +369,13 @@ class FeatureIDViewSet(viewsets.GenericViewSet):
         required=True,
         type=openapi.TYPE_STRING,
     )
+    oragnism_id = openapi.Parameter(
```
Comment (azneto) [inline machado/api/views.py:484]: typo: oragnism

File: machado/api/views.py [hunk 2]
```diff
@@ -369,6 +369,13 @@ class FeatureIDViewSet(viewsets.GenericViewSet):
...
+        "organism_id",
+        openapi.IN_QUERY,
+        description="organism id",
+        required=True,
+        type=openapi.TYPE_INTEGER,
+    )
 
     operation_summary = "Retrieve feature ID by accession"
     operation_description = operation_summary + "<br /><br />"
```
Comment (azneto) [inline machado/api/views.py:493]: The example below will not work anymore. Please, add MACHADO_EXAMPLE_ORGANISM.

**Review by azneto — APPROVED** [review]

**Comment by azneto** [comment]
It is looking good so far, but we need to thoroughly test it. The current tests will evaluate the functions, but the interfaces (loading tools, API, and web interface) will require manual testing. There's a [sample.tar.gz](https://github.com/lmb-embrapa/machado/blob/master/extras/sample.tar.gz) file that can be used for testing purposes. I suggest creating a new installation from scratch. 

In regard to JBrowseFeatureViewSet, you can create a new parameter (openapi.Parameter) for organism and modify get_serializer_context and get_queryset to filter by organism as well. The JBrowse configuration file [trackList.json.sample](https://github.com/lmb-embrapa/machado/blob/master/extras/trackList.json.sample) can be set to submit the organism:

```
         "query": {
            "organism": "Arabidopsis thaliana"
         }
```

**Comment by njbooher** [comment]
It's the first week of classes (which is a busy time for us) so there may be a delay. But I will try to get to this soon.

**Comment by njbooher** [comment]
i've tested this with the sample data using the loaders, jbrowse, and the api.

will test it with loading my real data later in the week.

failing test is related to https://github.com/lmb-embrapa/machado/commit/fdf04c64b8d9b04705ae684a012ad42c9b541867

**Comment by njbooher** [comment]
I have now tested this with my real data and it worked fine.

**Comment by azneto** [comment]
I'll review it and let you know. thx

**Comment by azneto** [comment]
Sorry it took so long. I've just created a brand new instance of Machado and went through every step from the documentation and it's running fine. The documentation still requires a few updates.

**Pull request merged**